SlowCheetah.JDT/JsonTransformationContextLogger.cs

```diff
@@ -0,0 +1,135 @@
+        /// <param name="extLogger">External logger to be used. Can be null.</param>
+        internal JsonTransformationContextLogger(string transformationFile, IJsonTransformationLogger extLogger)
+        {
+            this.externalLogger = extLogger;
+            this.TransformFile = transformationFile;
+        }
+
+        /// <summary>
+        /// Gets or sets the source file of the current transformation
+        /// </summary>
+        internal string SourceFile { get; set; } = "Source";
+
+        /// <summary>
+        /// Gets or sets the transformation file of the current transformation
+        /// </summary>
+        internal string TransformFile { get; set; } = "Transform";
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the logger has logged errrors
+        /// </summary>
+        internal bool HasLoggedErrors { get; set; }
+
+        /// <summary>
+        /// Logs an error from an internal exception
+        /// </summary>
+        /// <param name="exception">The exception to log</param>
+        /// <returns>True if the exception was logged</returns>
+        internal bool LogErrorFromException(Exception exception)
+        {
+            if (exception == null)
+            {
+                throw new ArgumentNullException(nameof(exception));
+            }
+
+            // Only log if an external logger has been provided
+            if (this.externalLogger != null)
+            {
+                this.HasLoggedErrors = true;
+
+                // First, attempt to convert to a JdtException that contains lineinfo and error location
+                JdtException jdtException = exception as JdtException;
+                if (jdtException != null)
+                {
+                    this.externalLogger.LogErrorFromException(jdtException, this.LocationPath(jdtException.Location), jdtException.LineNumber, jdtException.LinePosition);
+                }
+                else
+                {
+                    // JsonReader exceptions are caused by loading errors and contain file and line info
+                    JsonReaderException readerException = exception as JsonReaderException;
+                    if (readerException != null)
+                    {
+                        this.externalLogger.LogErrorFromException(readerException, readerException.Path, readerException.LineNumber, readerException.LinePosition);
+                    }
+                    else
+                    {
+                        // If the exception does not have any additional info on it
+                        this.externalLogger.LogErrorFromException(exception);
+                    }
+                }
+
+                // The exception has been logged
+                return true;
+            }
+            else
+            {
+                // The exception has not been logged
+                return false;
```

Comment:
As discussed in person, rethrow the exception here using `ExceptionDispathInfo.Capture(exception).Throw();` Then switch to a `void` return.